datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -509,6 +523,15 @@ impl GroupedHashAggregateStream {
             None
         };
 
+        // Check if we can enable the blocked optimization for `GroupValues` and `GroupsAccumulator`s.
+        let enable_blocked_group_states = maybe_enable_blocked_group_states(
+            &context,
+            group_values.as_mut(),
+            &mut accumulators,
+            batch_size,
```

Comment:
One thing to do is expose block size as a different configuration knob, and also find a good default value (Now it's using the same as batch size)

Reply:
> One thing to do is expose block size as a different configuration knob, and also find a good default value (Now it's using the same as batch size)

🤔  I think the `batch_size` is better equal to `n * block_size`? Otherwise, we need to `slice` it before emitting to next operator, and the `slice` is a actually a bit expansive.

Maybe we should consider how to expose the config option.